images/api-server-dev/Makefile

```diff
@@ -0,0 +1,30 @@
+KAMU_VERSION=$(shell cargo metadata --format-version 1 | jq -r '.packages[] | select( .name == "kamu-core") | .version')
+IMAGE_REPO=ghcr.io/kamu-data
+TARGET=x86_64-unknown-linux-gnu
+TAG=dev-$(shell git rev-parse --abbrev-ref HEAD | sed 's/\//-/g')
+
+
+################################################################################
+
+.PHONY: build
+build:
+	# cargo binstall cross
+	cd ../../ && RUSTFLAGS="" cross build -p kamu-api-server --release --target=$(TARGET)
+
+
+.PHONY: image
+image: build
+	mkdir -p ./tmp
+	cp ../../target/$(TARGET)/release/kamu-api-server tmp/kamu-api-server
+	docker build \
+		--build-arg KAMU_VERSION=$(KAMU_VERSION) \
+		-t $(IMAGE_REPO)/kamu-api-server:$(TAG) \
```

Comment:
I would like us to clearly separate prod and dev images and I think tag is insufficient because:
- Dev image uploads will relate to the same GH "package" as prod images
  - Harder for users to see the version history
  - If there will be ever image push notifications - we'll be spamming people with dev builds
- Makes it harder to clean up dev images over time - will need some tag-based regexes


In GHCR we cannot create additional namespaces like `ghcr.io/kamu-data-dev`, because namespace has to match the GH account. Should we instead change the image name to `kamu-dev-api-server` and use `kamu-dev` consistently as a prefix for dev builds?

Reply:
> I would like us to clearly separate prod and dev images and I think tag is insufficient because

Good idea, I agree -- также, я добавил тоже самое и для `kamu-cli` (https://github.com/kamu-data/kamu-cli/pull/969), please take a look.

> Should we instead change the image name to kamu-dev-api-server and use kamu-dev consistently as a prefix for dev builds?

Yes, done

Reply:
Thus, we have the following picture:
```
images
├── kamu-api-server
│   └── ...
├── kamu-api-server-with-data-mt
│   └── ...
├── kamu-dev-api-server
│   └── ...
├── kamu-oracle-provider
│   └── ...
└── Makefile
```

In that case, visual separation by namespace seems preferable (discursively).

Q: do you think, even in such a case, it's still worth sticking to two naming schemes?

Reply:
Consistency is good. Let's proceed with the proposed one.